components/follow_button/view.coffee

```diff
@@ -49,6 +51,8 @@ module.exports = class FollowButton extends Backbone.View
     @trigger 'click'
 
     unless @following
+      mediator.trigger 'clickFollowButton'
```

Comment:
So in the case of the artist page CTA, we just want to trigger the special event and return (and skip the whole auth/follow thing). Then the artist page CTA is setup to handle this mediator event and show the overlay.